.github/workflows/_deploy_cdk.yml

```diff
@@ -89,12 +93,26 @@ jobs:
           sparse-checkout: |
             config-oss/api-infra
       # checkout the internal repo to get configs - feature branch when "branching to staging"
+      - name: Determine which internal branch to checkout
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
I am getting this message in this part of the script:
gh: Not Found (HTTP 404)
Internal branch for checkout: develop
which is causing the downstream errors I am having, since it isn't finding the identically named internal feature branch. 

I think the issue is that this should be `GITHUB_TOKEN: ${{ secrets.SERVICE_PAT }}` since the automatically provided GITHUB_TOKEN in GitHub Actions has permissions scoped to the repository that contains the workflow. If the workflow is in a public repository which it is, this token won't have access to a separate private repository by default. I think?